hack/lib/golang.sh

```diff
@@ -775,12 +775,31 @@ kube::golang::build_binaries_for_platform() {
   fi
 
   for test in "${tests[@]:+${tests[@]}}"; do
-    local outfile testpkg
+    local outfile testpkg use_cgo
     outfile=$(kube::golang::outfile_for_binary "${test}" "${platform}")
     testpkg=$(dirname "${test}")
+    use_cgo="${CGO_ENABLED:-0}"
```

Comment:
should the default be 0 or 1?

Reply:
the fact that we only override to 0 for things in the statics list above makes me think the default should be 1 here, right?

Reply:
The tests failed when the default was `1`: https://storage.googleapis.com/kubernetes-jenkins/pr-logs/pull/115613/pull-kubernetes-cross/1623945772418469888/build-log.txt

So I changed the default to `0`, which seems to be right in the context of the test binaries.

Reply:
if we're defaulting to 0, then setting to 0 below specifically for linux/arm doesn't seem like it's doing anything, right?

Reply:
we have kubelet and e2e_node tests that use cgo, so turning this off is short-circuiting those tests, right?

Reply:
`-race`requires CGO to 1

Reply:
Ugh, yeah means we loose some tests when disabling CGO. :/

Reply:
it simple does not work, 
```
CGO_ENABLED=0 go test -race .
go: -race requires cgo; enable cgo by setting CGO_ENABLED=1
```
I don't know if those tests set that variable later, it was more just a FYI

Reply:
looks like race isn't supported on linux/arm anyway... I think defaulting CGO_ENABLED to "" would be more reasonable, though we'd still lose those two cgo tests on linux/arm (though I'm not even sure why they need cgo, honestly)

Reply:
bisected this to the unified build getting enabled in go1.20, details in https://github.com/golang/go/issues/58425#issuecomment-1426346686

alternative workaround in https://github.com/kubernetes/kubernetes/pull/115692